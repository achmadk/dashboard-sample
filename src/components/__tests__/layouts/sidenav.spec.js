import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import chai from 'chai'

import SidenavContainer, {
  sidenavComponent
} from '@/components/layouts/sidenav'
import SidenavComponent from '@/components/layouts/sidenav/sidenav'
import ProviderStore from '@/../test/provider-store'

const props = {
  enqueueSnackbar: jest.fn()
}

describe('sidenav Layout Container', () => {
  it('render correctly', () => {
    const wrapper = shallow(
      <ProviderStore>
        <SidenavContainer {...props} />
      </ProviderStore>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('return layout component successfully', () => {
    const result = sidenavComponent()
    chai.expect(result).to.exist
  })
})

describe('sidenav Layout Component', () => {
  it('render correctly', () => {
    const wrapper = shallow(
      <ProviderStore>
        <SidenavComponent {...props} />
      </ProviderStore>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
