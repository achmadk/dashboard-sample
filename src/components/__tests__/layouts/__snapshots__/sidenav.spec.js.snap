// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidenav Layout Component render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectFunction
    enqueueSnackbar={[MockFunction]}
  />
</Provider>
`;

exports[`sidenav Layout Container render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <SidenavContainer
    enqueueSnackbar={[MockFunction]}
  />
</Provider>
`;
